management/nodes_test.go

```diff
@@ -43,6 +43,51 @@ func TestNodeRegister(t *testing.T) {
 			}
 		})
 
+		t.Run("Reregister with different node name (no force - should fail)", func(t *testing.T) {
+			nodeName := pmmapitests.TestString(t, "node-name-for-all-fields")
+			nodeID, pmmAgentID := registerGenericNode(t, node.RegisterNodeBody{
+				NodeName: nodeName,
+				NodeType: pointer.ToString(node.RegisterNodeBodyNodeTypeGENERICNODE),
+				Address:  "node-address",
+				Region:   "region",
+			})
+			defer pmmapitests.RemoveNodes(t, nodeID)
+			defer removePMMAgentWithSubAgents(t, pmmAgentID)
+
+			body := node.RegisterNodeBody{
+				NodeName: nodeName,
+				NodeType: pointer.ToString(node.RegisterNodeBodyNodeTypeGENERICNODE),
+				Address:  "node-address",
+				Region:   "region",
+			}
+			params := node.RegisterNodeParams{
+				Context: pmmapitests.Context,
+				Body:    body,
+			}
+			_, err := client.Default.Node.RegisterNode(&params)
+			assert.Equal(t, err.(pmmapitests.ErrorResponse).Code(), 409)
```

Comment:
Can we check certain error message with assert.EqualError? I'm sure @AlekSi will request it

Reply:
The error includes de hostname and since it is variable, I choose to not to test it.
My first attempt was to use the already existing test helpers which use the error message but it was including my local hostname so I decided to check only for the error code.

Reply:
Currently, we have this helper: https://github.com/Percona-Lab/pmm-api-tests/blob/1213b16c56da082fb2daef92dc2e2a2be3b4ba9b/helpers.go#L39-L63

We probably should make another version with regular expressions like one we have in pmm-managed: https://github.com/percona/pmm-managed/blob/90c834b8e7f8cab179352075c98cef540bf31122/utils/tests/asserts.go#L42-L53

Reply:
:+1:

Reply:
So, we have `nodeName` variable and we can use it.